### PR TITLE
Move spring-cloud-azure-native-configuration to sdk/spring-experimental folder

### PR DESCRIPTION
**Pull request by saragluna**
Since `spring-native` is still an experimental project and the `spring-cloud-azure-native-configuration` won't be able to GA until the `spring-native` is GAed, so this PR will create a separate `sdk/spring-experimental` folder and further experimental projects will stay in this folder and be promoted to the `sdk/spring` folder once it gets GAed.

### CODE REVIEW

File: eng/scripts/generate_from_source_pom.py [hunk 1]
```diff
@@ -267,7 +267,9 @@ def is_spring_child_pom(tree_root: ET.Element):
     group_id_node = element_find(tree_root, 'groupId')
     artifact_id_node = element_find(tree_root, 'artifactId')
     return not group_id_node is None and group_id_node.text == 'com.azure.spring' \
-           and not artifact_id_node is None and artifact_id_node.text != 'spring-cloud-azure' # Exclude parent pom to fix this error: "Project is duplicated in the reactor"
+           and not artifact_id_node is None \
+           and (artifact_id_node.text != 'spring-cloud-azure' \
+                or artifact_id_node.text != 'spring-cloud-azure-experimental') # Exclude parent pom to fix this error: "Project is duplicated in the reactor"
```
Comment (jialigit) [inline eng/scripts/generate_from_source_pom.py]: Can add some comments explaining what is a Spring Library?

And the statement,  `not <var> is None` can be replaced with ` <var> is not None`?

File: sdk/spring-experimental/spring-cloud-azure-native-configuration/CHANGELOG.md [hunk 2]
```diff
@@ -4,9 +4,6 @@
 
 ### Features Added
 
-<!-- 
-NOTE: This changelog entry should be added back to Features Added section of the 'spring/CHANGELOG.md' when the native configuration is ready to release the beta version, and referenced here.
--->
 - Add library `spring-cloud-azure-native-configuration` to support `Spring Native` for compiling Spring Cloud Azure applications to native executables [#28053](https://github.com/Azure/azure-sdk-for-java/issues/28053), [#28158](https://github.com/Azure/azure-sdk-for-java/pull/28158).
```
Comment (jialigit) [inline sdk/spring-experimental/spring-cloud-azure-native-configuration/CHANGELOG.md:7]: `to native executables` seems not needed. It is not correct in grammar.

**Comment by jialigit** [comment]
> Since `spring-native` is still an experimental project and the `spring-cloud-azure-native-configuration` won't be able to GA until the `spring-native` is GAed, so this PR will create a separate `sdk/spring-experimental` folder and further experimental projects will stay in this folder and be promoted to the `sdk/spring` folder once it gets GAed.

Can you explain why we don't use a feature branch here? Adding an extra folder called "experimental" looks ugly really.

File: eng/scripts/generate_from_source_pom.py (hunk 1 above)
Comment (saragluna) [reply eng/scripts/generate_from_source_pom.py]: not the scope of this PR, please help create a pr for that.

File: sdk/spring-experimental/spring-cloud-azure-native-configuration/CHANGELOG.md (hunk 2 above)
Comment (saragluna) [reply sdk/spring-experimental/spring-cloud-azure-native-configuration/CHANGELOG.md:7]: how about checking this https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/#overview.

**Comment by saragluna** [comment]
> > Since `spring-native` is still an experimental project and the `spring-cloud-azure-native-configuration` won't be able to GA until the `spring-native` is GAed, so this PR will create a separate `sdk/spring-experimental` folder and further experimental projects will stay in this folder and be promoted to the `sdk/spring` folder once it gets GAed.
> 
> Can you explain why we don't use a feature branch here? Adding an extra folder called "experimental" looks ugly really.

The SDK repo chooses multi-folders over multiple branches, we are following this guideline.

**Review by moarychan — APPROVED** [review]
LGTM. These changes should be merged first, and need to improve the CI pipeline job for the coming release.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**